packages/providers/src/utils/change-case/functions.ts

```diff
@@ -0,0 +1,298 @@
+// Regexps involved with splitting words in various case formats.
+const SPLIT_LOWER_UPPER_RE = /([\p{Ll}\d])(\p{Lu})/gu;
+const SPLIT_UPPER_UPPER_RE = /(\p{Lu})([\p{Lu}][\p{Ll}])/gu;
+
+// Used to iterate over the initial split result and separate numbers.
+const SPLIT_SEPARATE_NUMBER_RE = /(\d)\p{Ll}|(\p{L})\d/u;
+
+// Regexp involved with stripping non-word characters from the result.
+const DEFAULT_STRIP_REGEXP = /[^\p{L}\d]+/giu;
+
+// The replacement value for splits.
+const SPLIT_REPLACE_VALUE = '$1\0$2';
+
+// The default characters to keep after transforming case.
+const DEFAULT_PREFIX_SUFFIX_CHARACTERS = '';
+
+/**
+ * Supported locale values. Use `false` to ignore locale.
+ * Defaults to `undefined`, which uses the host environment.
+ */
+export type Locale = string[] | string | false | undefined;
+
+/**
+ * Options used for converting strings to pascal/camel case.
+ */
+export interface IPascalCaseOptions extends IOptions {
+  mergeAmbiguousCharacters?: boolean;
+}
+
+/**
+ * Options used for converting strings to any case.
+ */
+export interface IOptions {
+  locale?: Locale;
+  split?: (value: string) => string[];
+  /** @deprecated Pass `split: splitSeparateNumbers` instead. */
+  separateNumbers?: boolean;
+  delimiter?: string;
+  prefixCharacters?: string;
+  suffixCharacters?: string;
+  keyCaseTransformer?: (key: string) => string;
+}
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** This interface should be declared above `IPascalCaseOptions` so that the types read like a book from top to bottom. I had to scroll back up to the top of the file looking for imports before I realised the type was below.

This makes for easier to read and more maintainable code.‏